src/server/node_manager.c

```diff
@@ -1118,11 +1118,11 @@ shallow_vnode_dup(struct pbsnode *vnode)
 	/*
 	 * Copy vnode elements (same order as "struct pbsnode" element definition)
 	 */
-	if (vnode_dup->nd_moms) {
-		/* free the initialize_pbsnode() allocation before assigning */
-		free(vnode_dup->nd_moms);
-		vnode_dup->nd_moms = NULL;
-	}
+	/* free the initialize_pbsnode() allocation before assigning */
+
+	free(vnode_dup->nd_name);
+	vnode_dup->nd_name = vnode->nd_name;
```

Comment:
vnode_dup->nd_name is already set to strdup(vnode->nd_name) from the call to initialize_pbsnode(), why free it and set it again (and without strdup)?

Reply:
The function does a shallow copy of the existing node. The earlier strdup is required so that we can call free_pnode on failure which would otherwise free the original copy.

Reply:
ok .. why not just pass NULL to initialize_pbsnode() for the vnode name argument?

Reply:
that should work, thanks!

Reply:
I've addressed the comments. Please have a look.